src/inmanta/agent/executor.py

```diff
@@ -141,9 +199,31 @@ def __eq__(self, other: object) -> bool:
             return False
         return (
             self.agent_name == other.agent_name
-            and self.blueprint.generate_blueprint_hash() == other.blueprint.generate_blueprint_hash()
+            and self.agent_uri == other.agent_uri
+            and self.blueprint.blueprint_hash() == other.blueprint.blueprint_hash()
         )
 
+    def identity(self) -> str:
+        return self.agent_name + self.agent_uri + self.blueprint.blueprint_hash()
+
+
+@dataclass(frozen=True)
+class ResourceInstallSpec:
```

Comment:
it packs a resource type and version with the blueprint, so in doing that, it has its relevance.
Mostly for backward compatibility, indeed. It is required to make the `in_process_executor`'s manager class work correctly.  


But if we want to have two implemenations that are interchangeable, we currently need it. 

(it being a class also makes it easier to strip it our later I think)

Reply:
I don't think I was sufficiently clear in what I meant. I wasn't suggesting to drop it, I see why we need it. I'm just wondering if it wouldn't be a better fit in `inmanta.agent.agent` since it has nothing to do intrinsically with the `Executor`.

But I think I overlooked that it is part of the signature of `get_executor` therefore it makes sense to have it here. So scratch this comment.